compiler/lcalc/to_ocaml.ml

```diff
@@ -158,10 +169,16 @@ let rec format_typ (fmt : Format.formatter) (typ : Dcalc.Ast.typ Pos.marked) : u
       Format.fprintf fmt "@[<hov 2>(%a)@]"
         (Format.pp_print_list
            ~pp_sep:(fun fmt () -> Format.fprintf fmt "@ *@ ")
-           (fun fmt t -> Format.fprintf fmt "%a" format_typ_with_parens t))
+           format_typ_with_parens)
         ts
   | TTuple (_, Some s) -> Format.fprintf fmt "%a" format_struct_name s
-  | TEnum (_, e) -> Format.fprintf fmt "%a" format_enum_name e
+  | TEnum (ts, e) when D.EnumName.compare e Ast.option_enum = 0 ->
```

Comment:
What are you printing exactly here for the option enum type?

Reply:
we arrived to the conclusion that the implementation is not enough specific to the use case. The ocaml printer have been modified in accordance to this.